### PR TITLE
fix: model.fit metric not collected issue.

### PR DESCRIPTION
**Pull request by Genesis929**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: bigframes/ml/core.py [hunk 1]
```diff
@@ -273,6 +273,8 @@ def _create_model_ref(
     def _create_model_with_sql(self, session: bigframes.Session, sql: str) -> BqmlModel:
         # fit the model, synchronously
         _, job = session._start_query_ml_ddl(sql)
+        if session._metrics is not None:
+            session._metrics.count_job_stats(job)
```
Comment (shobsi) [inline bigframes/ml/core.py:277]: Should we not do this inside _start_query_ml_ddl? That way all ML DDLs will be accounted in the session metrics

File: tests/system/large/ml/test_linear_model.py [hunk 2]
```diff
@@ -31,8 +31,14 @@ def test_linear_regression_configure_fit_score(penguins_df_default_index, datase
         ]
     ]
     y_train = df[["body_mass_g"]]
+
+    start_execution_count = df._block._expr.session._metrics.execution_count
+
     model.fit(X_train, y_train)
```
Comment (shobsi) [inline tests/system/large/ml/test_linear_model.py:37]: I think we have the opportunity to make this work work all ML APIs - fit, score, predict, transform, .... We should write such tests for all APIs by writing the metrics collection logic in more central place in the code.

Comment (Genesis929) [reply tests/system/large/ml/test_linear_model.py:37]: score, predict works, seems internally is calling other functions.

File: bigframes/ml/core.py (hunk 1 above)
Comment (Genesis929) [reply bigframes/ml/core.py]: Updated, and added an assertion in test for register.

File: bigframes/session/__init__.py [hunk 3]
```diff
@@ -1373,7 +1373,14 @@ def _start_query_ml_ddl(
         # https://cloud.google.com/bigquery/docs/customer-managed-encryption#encrypt-model
         job_config.destination_encryption_configuration = None
 
-        return bf_io_bigquery.start_query_with_client(self.bqclient, sql, job_config)
+        results_iterator, query_job = bf_io_bigquery.start_query_with_client(
```
Comment (shobsi) [inline bigframes/session/__init__.py:1376]: I see `start_query_with_client` already has the stats update

https://github.com/googleapis/python-bigquery-dataframes/blob/fd06d31f5e7edab3ce3af9a4e3e4415a4463ee5d/bigframes/session/_io/bigquery/__init__.py#L245-L246

so feels like we could be adding double counting

File: tests/system/large/ml/test_linear_model.py [hunk 4]
```diff
@@ -31,8 +31,14 @@ def test_linear_regression_configure_fit_score(penguins_df_default_index, datase
...
 
+    end_execution_count = df._block._expr.session._metrics.execution_count
+    assert end_execution_count - start_execution_count == 2
```
Comment (shobsi) [inline tests/system/large/ml/test_linear_model.py:40]: Worth adding a comment why fit does 2 queries

File: tests/system/large/ml/test_linear_model.py (hunk 2 above)
Comment (shobsi) [reply tests/system/large/ml/test_linear_model.py:37]: Good to know, we could just pick one or two of the existing tests that does fit, score, predict, register and transform and assert everywhere that we are counting stats for those operations.

File: bigframes/session/__init__.py (hunk 3 above)
Comment (Genesis929) [reply bigframes/session/__init__.py]: It's not double counting, but yes here we can send the metric into the start_query_with_client, thanks for point that out.

File: tests/system/large/ml/test_linear_model.py (hunk 2 above)
Comment (Genesis929) [reply tests/system/large/ml/test_linear_model.py:37]: Test Added.

**Review by shobsi — APPROVED** [review]
